### PR TITLE
dont merge - its extension is New History txns

### PR DESCRIPTION
**Pull request by aadarsh012**

### CODE REVIEW

**Comment by aadarsh012** [comment]
image

File: src/utils/approveMultisigTransfer.ts [hunk 1]
```diff
@@ -183,6 +185,7 @@ export async function approveMultisigTransfer ({ amount, api, approvingAddress,
 									note,
 									to: recipientAddress || ''
 								});
+								console.log('transaction added to history');
```
Comment (alphainfinitus) [inline src/utils/approveMultisigTransfer.ts:188]: lingering console log

File: src/types.ts [hunk 2]
```diff
@@ -242,17 +243,15 @@ export interface ITransaction {
 	id: string;
 	token: string;
 	amount_usd: number;
-	amount_token: number | number[];
+	amount_token: string;
```
Comment (alphainfinitus) [inline src/types.ts:246]: please confirm this change

File: cloud-functions/src/index.ts [hunk 3]
```diff
@@ -2825,3 +2827,49 @@ export const getSharedAddressBook = functions.https.onRequest(async (req, res) =
...
+				encodedMultisigAddress,
+				network,
+				Number(limit),
+				Number(page),
+				firestoreDB
+			);
+
+			if (historyItemsError || !historyItemsArr) return res.status(400).json({ error: historyItemsError || responseMessages.queue_fetch_error });
+
+			res.status(200).json({ data: historyItemsArr, count });
```
Comment (alphainfinitus) [inline cloud-functions/src/index.ts:2857]: count should be returned inside data for consistency.
data and error should be the only things returned from the cloud functions

File: src/components/Transactions/History/Transaction.tsx [hunk 4]
```diff
@@ -15,25 +15,58 @@ import { FIREBASE_FUNCTIONS_URL } from 'src/global/firebaseFunctionsUrl';
...
+
+		const { data, error } = decodeCallData(callData, api);
+		if(error || !data) return;
+
+		if(data?.extrinsicCall?.hash.toHex() !== callHash) {
+			return;
+		}
+
+		setDecodedCallData(data.extrinsicCall?.toJSON());
+		console.log(data.extrinsicCall?.toJSON());
```
Comment (alphainfinitus) [inline src/components/Transactions/History/Transaction.tsx:52]: lingering console log

File: src/types.ts (hunk 2 above)
Comment (aadarsh012) [reply src/types.ts:246]: Yes, this is handled everywhere, there wont be any problem for older txns too

File: src/components/Transactions/History/index.tsx [hunk 5]
```diff
@@ -29,6 +30,7 @@ const History: FC<IHistory> = ({ loading, setLoading, refetch }) => {
 	const userAddress = localStorage.getItem('address');
 	const signature = localStorage.getItem('signature');
 	const { activeMultisig, multisigAddresses } = useGlobalUserDetailsContext();
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (alphainfinitus) [inline src/components/Transactions/History/index.tsx:33]: please check if this comment is needed or if the variable is not being used

File: src/components/Transactions/History/index.tsx [hunk 6]
```diff
@@ -53,21 +55,28 @@ const History: FC<IHistory> = ({ loading, setLoading, refetch }) => {
...
+				const getMultisigHistoryTransactions = await fetch(`${FIREBASE_FUNCTIONS_URL}/getMultisigHistory`, {
+					body: JSON.stringify({
+						limit: multisig.proxy ? 5 : 10,
+						multisigAddress: multisig?.address,
+						page: currentPage
+					}),
+					headers: firebaseFunctionsHeader(network),
+					method: 'POST'
+				});
+				const { data: multisigTransactions, error: multisigError, count:multisigTransactionsCount } = await getMultisigHistoryTransactions.json() as { data: ITransaction[], error: string, count: number };
```
Comment (alphainfinitus) [inline src/components/Transactions/History/index.tsx:68]: count should be inside data

File: src/components/Transactions/History/index.tsx [hunk 7]
```diff
@@ -53,21 +55,28 @@ const History: FC<IHistory> = ({ loading, setLoading, refetch }) => {
...
+					const getProxyHistoryTransactions = await fetch(`${FIREBASE_FUNCTIONS_URL}/getMultisigHistory`, {
+						body: JSON.stringify({
+							limit: 10 - multisigTransactions.length,
+							multisigAddress: multisig.proxy,
+							page: currentPage
+						}),
+						headers: firebaseFunctionsHeader(network),
+						method: 'POST'
+					});
+					const { data: proxyTransactions, error: proxyError, count:proxyTransactionsCount } = await getProxyHistoryTransactions.json() as { data: ITransaction[], error: string, count: number };
```
Comment (alphainfinitus) [inline src/components/Transactions/History/index.tsx:79]: count should be inside data

File: cloud-functions/src/utlils/getHistoryTransactions.ts [hunk 8]
```diff
@@ -0,0 +1,117 @@
+import axios from 'axios';
+import { ITransaction } from '../types';
+import { SUBSCAN_API_HEADERS } from '../constants/subscan_consts';
+import { responseMessages } from '../constants/response_messages';
+import dayjs from 'dayjs';
+import { firestore } from 'firebase-admin';
+
+interface IResponse {
+	error?: string | null;
+	data: ITransaction[];
```
Comment (alphainfinitus) [inline cloud-functions/src/utlils/getHistoryTransactions.ts:10]: `data: {
 transactionData: ITransaction[]
 count:  number
}`

**Pull request closed without merging**